### PR TITLE
Sscs 4290 send additional evidence

### PR DESCRIPTION
**Pull request by iguannago**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/domain/email/EmailAttachment.java [hunk 1]
```diff
@@ -23,9 +23,17 @@ public static EmailAttachment pdf(byte[] content, String fileName) {
...
-                new ByteArrayResource(content),
-                "application/json",
-                fileName
+            new ByteArrayResource(content),
+            "application/json",
+            fileName
+        );
+    }
+
+    public static EmailAttachment file(byte[] content, String fileName) {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/domain/email/EmailAttachment.java]: Please, ensure this is covered by junit test.

File: src/main/java/uk/gov/hmcts/reform/sscs/domain/email/EmailAttachment.java [hunk 2]
```diff
@@ -23,9 +23,17 @@ public static EmailAttachment pdf(byte[] content, String fileName) {
 
     public static EmailAttachment json(byte[] content, String fileName) {
         return new EmailAttachment(
-                new ByteArrayResource(content),
-                "application/json",
-                fileName
+            new ByteArrayResource(content),
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/domain/email/EmailAttachment.java]: ensure junit coverage here.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java [hunk 3]
```diff
@@ -39,4 +46,16 @@ public UploadResponse upload(List<MultipartFile> files) {
         }
     }
 
+    public byte[] download(String documentId) {
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: ensure junit coverage here

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java [hunk 4]
```diff
@@ -39,4 +46,16 @@ public UploadResponse upload(List<MultipartFile> files) {
...
+        String serviceAuthorization = authTokenGenerator.generate();
+
+        try {
+            ResponseEntity<Resource> responseEntity =  documentDownloadClientApi.downloadBinary(DUMMY_OAUTH_2_TOKEN, serviceAuthorization, documentId);
```
Comment (iguannago) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: change the name of the token, please. Also it's not OAUTH, that's the service to service token :)

Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: s2s token we usually call it

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java (hunk 3 above)
Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: There are already several tests in uk.gov.hmcts.reform.sscs.service.EvidenceManagementServiceTest which cover this.

File: src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java [hunk 5]
```diff
@@ -33,10 +40,22 @@ public UploadResponse upload(List<MultipartFile> files) {
...
+        String serviceAuthorization = authTokenGenerator.generate();
+
+        try {
+            ResponseEntity<Resource> responseEntity =  documentDownloadClientApi.downloadBinary(S2S_TOKEN, serviceAuthorization, documentId);
+
+            ByteArrayResource resource = (ByteArrayResource) responseEntity.getBody();
+            return resource.getByteArray();
+        } catch (HttpClientErrorException httpClientErrorException) {
+            throw new UnsupportedDocumentTypeException(httpClientErrorException);
+        }
```
Comment (sathishbabuj) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: Log the exception before throwing back to the client. We had an issue on friday and we were not able correlate the failure from SYA to TCA API.

Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: Will do, thanks.

Comment (arcotc) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/EvidenceManagementService.java]: Done

**Comment by iguannago** [comment]
Approved!

File: build.gradle [hunk 6]
```diff
@@ -10,7 +10,7 @@ plugins {
 }
 
 group 'uk.gov.hmcts.reform'
-version '0.0.85'
+version '0.0.86-CANDIDATE'
```
Comment (jackmaloney) [inline build.gradle]: Could we change the version number so it doesn't contain Candidate?

File: src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsService.java [hunk 7]
```diff
@@ -41,33 +46,53 @@ public RoboticsService(
...
+        addAdditionEvidenceAttachments(additionalEvidence, attachments);
+        emailService.sendEmail(
+                roboticsEmailTemplate.generateEmail(
+                        appellantUniqueId,
+                        attachments
+                )
+        );
+    }
+
+    private void addAdditionEvidenceAttachments(Map<String, byte[]> additionalEvidence, List<EmailAttachment> attachments) {
```
Comment (jackmaloney) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsService.java]: typo in the method name 'Addition' should be 'Additional'

File: build.gradle (hunk 6 above)
Comment (iguannago) [reply build.gradle]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsService.java (hunk 7 above)
Comment (iguannago) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/RoboticsService.java]: done

**Pull request merged**